### ISSUE TITLE
Retranslation (translating sr3 Documentation into French)

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#539]
We have v2 Documentation here in English:

* English: https://github.com/MetPX/sarracenia/tree/v2_stable/doc
* French: https://github.com/MetPX/sarracenia/tree/v2_stable/doc/fr

and the English documentation for sr3 is here:

* https://github.com/MetPX/sarracenia/tree/v03_wip/docs/source

We need to figure out:

1. how to map French language documentation into the sphinx generated tree (so readers can switch between Eng and Fr)
  - Currently we aare trying a sub-tree approach, will require remaking/translating the sidebar contents. (can you have a per-dir sidebar?)
2. Prioritize items for translation (e.g. historical documents last.)
3. probably map it out into a series of issues or a tick-box list so we have a collective status on the work.  and perhaps assigne parts of it to multiple Francophones.
4. Actual translation of the current English documentation, likely  "grepping" through v2 English documentation... then looking at the French equivalents. and doing copy paste for many things, but it might be simpler to just re-translate.  Can use v2/fr docs at least for inspiration.  

We could investigate multiple approaches to getting things translated:
* get translation professionally done.
* figure out the pythonic way to internationalize, and structure things for that.

## TODO List:
- [x] Home.rst
- [x] index.rst (James to explore per-dir index/sidebar)
- **Reference** This is on the branch **v03_wip_doc_dr**
   - [x] index.rst
   - [x] sr3.1.rst **----> Clea**
   - [x] sr3_cpump.1.rst **----> Simon**
   - [x] sr3_credentials.7.rst **------> Clea**
   - [x] sr3_options.7.rst **-----> Clea**
   - [x] sr3_post.1.rst **----> Simon**
   - [x] sr_post.7.rst **----- Clea**
- **How2Guides**
   - [x] Admin.rst (v2: doc/fr/Admin.rst)
   - [x] Admin_Rabbit_Addendum.rst
   - [x] docker.rst
   - [x] Email_Ingesting_With_Sarracenia.rst
   - [x] FlowCallbacks.rst
   - [x] Hydro_Examples.rst
   - [x] index.rst
   - [x] source.rst  (v2: doc/fr/source.rst )    (English is not sr3 yet... still v2, so straight copy but then we need to update both.)
   - [x] subscriber.rst (v2: doc/fr/subscriber.rst )  (just need to convert examples to sr3.)
   - [x] UPGRADING.rst
   - [x] v2ToSr3.rst
- **Tutorials**  **-------> Simon**
   - [x] 1_CLI_introduction.ipynb
   - [x] 2_CLI_with_flowcb_demo.ipynb
   - [x] 3_api_flow_demo.ipynb
   - [x] 4_api_moth_sub_demo.ipynb
   - [x] 5_api_moth_post_demo.ipynb
   - [x] hello_world_server.txt
   - [x] index.rst
   - [x] Install.rst
   - [x] README.md
   - [x] requirements.txt
   - [x] Setup_a_local_subscriber.rst
   - [x] Setup_a_remote_subscriber.rst
   - [x] Windows.rst
- **Explanation** **-------> Clea (I can just work on all the Explanation stuff this week** 
   - [x] CommandLineGuide.rst (copy paste from doc/fr/sr_subscribe.1.rst near the beginning.)
   - [x] Concepts.rst (exists in v2) 
   - [x] DeploymentConsiderations.rst  (exists in v2) 
   - [x] DetectFileReady.rst (v2: doc/fr/sr_watch.1.rst#strategie de detection de fichiers. )
   - [x] FileCompletion.rst  (v2: doc/fr/sr_subscribe.1.rst#Assurer la livraison ) 
   - [x] Glossary.rst
   - [x] index.rst
   - [x] Overview.rst   (v2: doc/fr/sarra.rst ) 
   - [x] SarraPluginDev.rst  (completely different from v2, so must translate.)
   - [x] sftps.rst
   - [x] **History**
      - [x] deployment_2018.rst
      - [x] Evolution.rst
      - [x] HPC_Mirroring_Use_Case.rst
      - [x] index.rst
      - [x] mesh_gts.rst
      - [x] messages_v01.rst
      - [x] messages_v02.rst
      - [x] messages_v03.rst
      - [x] sr3_Announcement.rst
- **Contribution**
   - [x] AMQPprimer.rst
   - [x] BasicIdea.rst
   - [x] deltas.rst
   - [x] Design.rst
   - [x] Development.rst
   - [x] Documentation.rst
   - [x] Evolution.rst
   - [x] index.rst
   - [x] man_page_template.rst
   - [x] mqtt_issues.rst
   - [x] on_part_assembly.rst
   - [x] v03.rst

**Comment by MagikEh** [comment]
Closing #58 in favor of this issue.

**Comment by MagikEh** [comment]
`/doc/fr/` still exists, translators can use it as reference material to pull into the updated `/docs/source/fr/` dir

**Comment by MagikEh** [comment]
# Sarra Docu Meeting Notes:
- In browser translate mangles the technical docs (autogenerated files get function names and raw code translated...)
- There's a TODO list in the OP in order of precedence (top to bottom)

### Rough Translation Process 
- Claim a file to translate in the OP (put your name beside it)
- Copy the file from `/docs/source/*` to `/docs/source/fr/*`
- translate it!
- When finished, check off the item in the OP

**Comment by cleaaum** [comment]
Maybe a few thing we should agree on for translation purposes:

1. should flow be translated to flux
2. should component be translated to composant, and if yes, should we make it masculine or feminine (compsant/composante)
3. For "tree structure" the internet suggests the term "arborescence" .. does that make sense or do we have a better word.
4. How should we say poll in french

**Comment by MagikEh** [comment]
### Weekly Dev Meeting Notes:
- @cleaaum made [a glossary in docs/source/source/fr/](https://github.com/MetPX/sarracenia/blob/v03_wip_doc_dr/docs/source/source/fr/glossary.rst) for some terms/words that we may not want to translate directly as to maintain a technical wordset.

**Comment by petersilva** [comment]
priorities: 

 - index.rst in each directory (pointing to all English resources.)
 - documents for which French versions likely already easily copy/pasted.
 - so Explanation has the most of those. Probable best as next target.  ( lower priorities: minus: sftps.rst, and minus: History )
 - after that: Tutorials.

**Comment by petersilva** [comment]
I had a look at the current branch... Good Stuff!

* weird that fr/ is in docs/source/source/fr     I would think one source is enough... 
* in sr3.1.rst  " Sr3 permet à l'utilisateur de *commencer*, arrêter, et se renseigner sur
le statut d'un flux de sarracenia déjà configuré" ... shouldn't that be *démarrer"  ?
*  (in sr3_options.7.rst) "floating point" is translated as "nombre a virgule flottante"  ... which is cool... language wise... but the decimal marker really is a point, and not a comma... I noticed the example used 0,001   so I changed it to a point... but... language-wise should we say something like "nombre à virgule flottante, avec point comme marquer de décimale" ?  something like that...
* there is some weird downcasing going on, I caught a few... but "permdefault" is wrong... it's case sensitive, and the option is called permDefault   ... I think that is just a one-off... rest looks good.
* does "publicités"  make any sense in French? I know I used all kinds of different words for the same thing in English:  messages, notifications, advertisements, publications,   ... hmm... publicités sounds more commercial in French... I wanted to use dépêches in French, but the francophones vetoed me in v2 stuff.... 
* also weird... article? is used in sr_post.1.rst? hmm...  really?
* for titles... lots of cases there were string substitutions made, but the underlines don´t match the title length anymorre... rst processors will get upset.
* for titles ... need  a blank line before and after... I had to add a bunch of them.

I committed a bunch of small things related to the bullets above in the branch.

I noticed most of the links for more information in the man pages pointed to Explanation, so that reenforces Explanation´s priority as the next target.

**Comment by petersilva** [comment]
@petersilva needs to merge this branch... and then we can keep going from where they left off.

**Comment by petersilva** [comment]
Done to the point that, in future, we just require both languages to contributions to the v03_wip branch.

**Comment by petersilva** [comment]
@cleaaum I saw the recent patches, you started on contributions... that's fine, but priority-wise the how2's (commentfaire?) are probably a higher priority.  That directory does not seem to be done.  I also noted only the first few lines of miseaniveau.rst are done... the rest is still English...
both higher priorities than anything in Contribution.

**Comment by cleaaum** [comment]
@petersilva ok I will continue with How2 tomorrow and leave Contribution alone for now. Cant remember if I worked on the Mise a Niveau but I will fix it tomorrow regardless. But yes it is normal if there is English sometimes. I copy over the whole English file and then change to the French bit by bit, but if I am done for the day, then I just commit what I have so far and finish the rest the next day! So not to worry about the English. It will be gone:)

**Comment by cleaaum** [comment]
FRENCH DOC UPDATE: I have offically finished all of the french documentation. There are for sure a few errors with formatting/links/terminology. I will try yo fix up as much as I can. But here are some things to watch out for for whoerver might be looking at doc next.
 1. Links to images need a new path. If an image isnt showing up, the path needs to be renamed to ../../img_path since the images were not copied in the fr directory.
 2. poll was never really translated and was left as poll in the doc. We can change it to interroger/sondage if prefered. 
 3. topic and suntopic were also left as it. We discussed in the last dev meeting that this should be changed to sujet and sous-sujet. Sometimes rubrique was used as well but that means theme and should be changed back to sujet.
 4. How2Guides was translated to comment faire. I did not find a better name for it.
